base/abstractarray.jl

```diff
@@ -1218,6 +1218,7 @@ end
 copymutable(itr) = collect(itr)
 
 zero(x::AbstractArray{T}) where {T<:Number} = fill!(similar(x, typeof(zero(T))), zero(T))
+zero(x::AbstractArray{S}) where {S<:Union{Missing, Number}} = fill!(similar(x, typeof(zero(S))), zero(S))
 zero(x::AbstractArray) = map(zero, x)
```

Comment:
we could.
My thoughts on that when i initially did this was that it wasn't worth doing.
Because in the greater context of why you are callign `zero` on a `AbstractArray`,
you are doing linear algebra.
And other linear algebra operations like `+` and scalar multiplication,  also error when you do them on an array containing any undefined elements.